pkg/apis/eventing/v1alpha1/trigger_types.go

```diff
@@ -76,6 +85,26 @@ type TriggerFilterSourceAndType struct {
 	Source string `json:"source,omitempty"`
 }
 
+// TriggerFilterCEL filters events by evaluating the expression with the Common
+// Expression Language runtime. An event passes the filter if the expression
+// evaluates to true.
+type TriggerFilterCEL struct {
+	// Expression is the CEL expression to evaluate. Required.
+	Expression string `json:"expression,omitempty"`
+	// ParseExtensions enables parsing of dynamic extensions attached to the event
```

Comment:
What is considered a dynamic extension? Is it anything that is not defined in the CloudEvents spec? 

If so, which version of the spec? If they only ever add extensions, then we can put a minimal version. But if they ever remove an attribute from a subsequent spec version, then if this is false, the user suddenly has a non-working filter, as that extension won't exist.

Reply:
Yes, anything not defined in the CloudEvents spec is considered a dynamic extension here. I should clarify that in the comment.

That said, practically the list of "official attributes" available in the CEL environment is defined here, not in the CloudEvents spec. If CloudEvents later removes or renames an attribute, we can keep the original available in the CEL environment to avoid breaking Triggers that previously had `parseExtensions` set to `false`.